oak-run/src/main/java/org/apache/jackrabbit/oak/run/FrozenNodeRefsByScanningCommand.java

```diff
@@ -0,0 +1,293 @@
+        OptionSpec<String> passwordOption = parser.accepts("password", "Password").withOptionalArg().defaultsTo("admin");
+
+        OptionSpec<String> testCreateRefPathOption = parser.accepts("testCreateRefPath", "FOR TESTING ONLY: path where to create a reference from")
+                .withOptionalArg();
+        OptionSpec<String> testCreateRefPropOption = parser.accepts("testCreateRefProp", "FOR TESTING ONLY: property name for create a reference")
+                .withOptionalArg();
+        OptionSpec<String> testCreateRefTypeOption = parser
+                .accepts("testCreateRefType", "FOR TESTING ONLY: property type: 'reference' or anything else for a plain String").withOptionalArg();
+        OptionSpec<String> testCreateRefUuidOption = parser.accepts("testCreateRefUuid", "FOR TESTING ONLY: uuid to use as the reference")
+                .withOptionalArg();
+
+        OptionSet options = opts.parseAndConfigure(parser, args);
+
+        System.out.println("Opening nodestore...");
+        NodeStoreFixture nodeStoreFixture = NodeStoreFixtureProvider.create(opts);
+        System.out.println("Nodestore opened.");
+
+        if (uuidscan(userOption, passwordOption, testCreateRefPathOption, testCreateRefPropOption, testCreateRefTypeOption, testCreateRefUuidOption,
+                options, nodeStoreFixture)) {
+            System.err.println("FAILURE: References (in any uuid formatted property value) to nt:frozenNode found.");
+            System.exit(1);
+        } else {
+            System.out.println("SUCCESS: No references (in any uuid formatted property value) to nt:frozenNode found.");
+        }
+    }
+
+    private boolean uuidscan(OptionSpec<String> userOption, OptionSpec<String> passwordOption, OptionSpec<String> testCreateRefPathOption,
+            OptionSpec<String> testCreateRefPropOption, OptionSpec<String> testCreateRefTypeOption, OptionSpec<String> testCreateRefUuidOption,
+            OptionSet options, NodeStoreFixture nodeStoreFixture) throws IOException {
+        List<FrozenNodeRef> list = new LinkedList<FrozenNodeRef>();
+
+        NodeStore nodeStore = nodeStoreFixture.getStore();
+        String user = userOption.value(options);
+        String password = passwordOption.value(options);
+
+        String createRefPath = testCreateRefPathOption.value(options);
+        String createRefProp = testCreateRefPropOption.value(options);
+        String createRefType = testCreateRefTypeOption.value(options);
+        String createRefUuid = testCreateRefUuidOption.value(options);
+
+        Closer closer = Utils.createCloserWithShutdownHook();
+        closer.register(nodeStoreFixture);
+        try {
+
+            System.out.println("Logging in...");
+            Session session = openSession(nodeStore, "crx.default", user, password);
+
+            if (createRefPath != null && createRefProp != null && createRefUuid != null) {
```

Comment:
I would move this part into a dedicated test class.

Reply:
Agree, much nicer. Done now at https://github.com/apache/jackrabbit-oak/pull/238/commits/fb8b330c4f248078bb2dce62cfb19ae90a50030c